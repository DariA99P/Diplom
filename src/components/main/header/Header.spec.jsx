import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '.';

describe('<Header />', () => {
  let wrapper;
  describe('On rendering', () => {
    let toggle;

    beforeEach(() => {
      toggle = jest.fn();
      wrapper = shallow(<Header toggle={toggle} />);
    });

    describe('When header is collapsed', () => {
      beforeEach(() => {
        wrapper.setProps({ collapsed: true });
      });

      it('should render toggle button on header', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('Should call handle toggle function when clicked', () => {
        wrapper.find('Icon').first().simulate('click');
        expect(toggle.mock.calls.length).toEqual(1);
      });
    });

    describe('When header is not collapsed', () => {
      beforeEach(() => {
        wrapper.setProps({ collapsed: false });
      });

      it('should not render toggle button on header', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
