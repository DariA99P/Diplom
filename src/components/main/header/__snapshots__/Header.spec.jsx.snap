// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> On rendering When header is collapsed should render toggle button on header 1`] = `
<Adapter
  className="templateHeader"
>
  <Row
    align="center"
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <div
      className="headerInfo"
    >
      <Icon
        className="triggerUnfold"
        onClick={[Function]}
        type="menu-unfold"
      />
      <withRouter(Breadcrumb)
        className="templateBreadcrumb"
        style={
          Object {
            "color": "var(--white, #FFFFFF)",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "color": "var(--white)",
          "fontSize": "20px",
        }
      }
    >
      VistaJet Microservice
      <Tag />
    </div>
    <div
      className="userInfo"
    >
      <Clock
        style={
          Object {
            "lineHeight": "1px",
          }
        }
      />
      <Dropdown
        mouseEnterDelay={0.15}
        mouseLeaveDelay={0.1}
        overlay={
          <Menu
            className=""
            focusable={false}
            prefixCls="ant-menu"
            theme="light"
          >
            <MenuItem>
              About
            </MenuItem>
          </Menu>
        }
        placement="bottomLeft"
        prefixCls="ant-dropdown"
        style={
          Object {
            "lineHeight": "1px",
          }
        }
        trigger={
          Array [
            "click",
          ]
        }
      >
        <span>
          <Icon
            type="down"
          />
        </span>
      </Dropdown>
    </div>
  </Row>
</Adapter>
`;

exports[`<Header /> On rendering When header is not collapsed should not render toggle button on header 1`] = `
<Adapter
  className="templateHeader"
>
  <Row
    align="center"
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <div
      className="headerInfo"
    >
      <withRouter(Breadcrumb)
        className="templateBreadcrumb"
        style={
          Object {
            "color": "var(--white, #FFFFFF)",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "color": "var(--white)",
          "fontSize": "20px",
        }
      }
    >
      VistaJet Microservice
      <Tag />
    </div>
    <div
      className="userInfo"
    >
      <Clock
        style={
          Object {
            "lineHeight": "1px",
          }
        }
      />
      <Dropdown
        mouseEnterDelay={0.15}
        mouseLeaveDelay={0.1}
        overlay={
          <Menu
            className=""
            focusable={false}
            prefixCls="ant-menu"
            theme="light"
          >
            <MenuItem>
              About
            </MenuItem>
          </Menu>
        }
        placement="bottomLeft"
        prefixCls="ant-dropdown"
        style={
          Object {
            "lineHeight": "1px",
          }
        }
        trigger={
          Array [
            "click",
          ]
        }
      >
        <span>
          <Icon
            type="down"
          />
        </span>
      </Dropdown>
    </div>
  </Row>
</Adapter>
`;
